tests/test_model_saving.py

```diff
@@ -161,5 +162,110 @@ def test_saving_right_after_compilation():
     os.remove(fname)
 
 
+@keras_test
+def test_loading_weights_by_name():
+    """
+    test loading model weights by name on:
+        - sequential model
+    """
+
+    # test with custom optimizer, loss
+    custom_opt = optimizers.rmsprop
+    custom_loss = objectives.mse
+
+    # sequential model
+    model = Sequential()
+    model.add(Dense(2, input_dim=3, name="rick"))
+    model.add(Dense(3, name="morty"))
+    model.compile(loss=custom_loss, optimizer=custom_opt(), metrics=['acc'])
+
+    x = np.random.random((1, 3))
+    y = np.random.random((1, 3))
+    model.train_on_batch(x, y)
+
+    out = model.predict(x)
+    old_weights = [layer.get_weights() for layer in model.layers]
+    _, fname = tempfile.mkstemp('.h5')
```

Comment:
I think the [`tempfile.mkstemp()`](https://docs.python.org/2/library/tempfile.html#tempfile.mkstemp) is a more robust way of generating temp files in a platform agnostic way (instead of `np.random`). Thoughts about replacing the other occurrences for consistency? (Or reverting this one?)